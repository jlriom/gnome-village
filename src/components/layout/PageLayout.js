import React from "react";
import PropTypes from "prop-types";

const PageLayout = ({ headerText, children }) => {
	return (
		<>
			<h2 className='my-3'>{headerText}</h2>
			<div>{children}</div>
		</>
	);
};

PageLayout.propTypes = {
	headerText: PropTypes.string,
	children: PropTypes.element
};

export default PageLayout;
