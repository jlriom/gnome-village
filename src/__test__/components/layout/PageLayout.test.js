import React from "react";
import PageLayout from "../../../components/layout/PageLayout";
import renderer from "react-test-renderer";

describe("<PageLayout />", () => {
    it("renders ok", () => {
        const tree = renderer.create(
            <PageLayout headerText={"header"} />
        );
        expect(tree).toMatchSnapshot();
    });
});

