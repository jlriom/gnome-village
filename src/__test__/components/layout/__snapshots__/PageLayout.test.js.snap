// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageLayout /> renders ok 1`] = `
Array [
  <h2
    className="my-3 display-4"
  >
    header
  </h2>,
  <hr />,
  <div />,
]
`;
